python/dask_cudf/dask_cudf/core.py

```diff
@@ -45,7 +45,11 @@ def optimize(dsk, keys, **kwargs):
 
 
 def finalize(results):
-    return cudf.concat(results)
+    if results and isinstance(
+        results[0], (cudf.DataFrame, cudf.Series, cudf.Index)
```

Comment:
Should we handle MultiIndex here as well? It doesn't subclass from Index AFAIK.

Reply:
I always forget that pesky multiindex :)